api/cpp/include/slint_image.h

```diff
@@ -11,7 +11,100 @@
+        : m_width(width), m_height(height), m_data(width * height)
+    {
+    }
+
+    /// Construct a SharedPixelBuffer by copying the data from the \a data array.
+    /// The array must be of size \a width * \a height
+    SharedPixelBuffer(uint32_t width, uint32_t height, const Pixel *data)
+        : m_width(width), m_height(height), m_data(data, data + (width * height))
+    {
+    }
+
+    /// Return the width of the buffer in pixels
+    uint32_t width() const { return m_width; }
+    /// Return the height of the buffer in pixels
+    uint32_t height() const { return m_height; }
+
+    /// Returns a const pointer to the first pixel of this buffer.
+    const Pixel *begin() const { return m_data.begin(); }
+    /// Returns a const pointer past this buffer.
+    const Pixel *end() const { return m_data.end(); }
+    /// Returns a pointer to the first pixel of this buffer.
+    Pixel *begin() { return m_data.begin(); }
+    /// Returns a pointer past this buffer.
+    Pixel *end() { return m_data.end(); }
+    /// Returns a const pointer to the first pixel of this buffer.
+    const Pixel *cbegin() const { return m_data.begin(); }
+    /// Returns a const pointer past this buffer.
+    const Pixel *cend() const { return m_data.end(); }
+
+    /// Compare two SharedPixelBuffer. They are considered equal if all their pixels are equal
+    bool operator==(const SharedPixelBuffer &other) const = default;
+
+private:
+    friend struct Image;
+    uint32_t m_width;
+    uint32_t m_height;
+    SharedVector<Pixel> m_data;
+};
+
 /// An image type that can be displayed by the Image element
+///
+/// You can construct Image objects from a path to an image file on disk, using
+/// Image::load_from_path().
+///
+/// Another typical use-case is to render the image content with C++ code.
+/// For this it’s most efficient to create a new SharedPixelBuffer with the known dimensions and
+/// pass the mutable slice to your rendering function. Afterwards you can create an Image using
+/// the constructor taking a SharedPixelBuffer.
+///
+/// The following example creates a 320x200 RGB pixel buffer and calls a function to draw a shape
+/// into it.
+/// ```c++
+/// slint::SharedPixelBuffer::<slint::Rgb8Pixel> pixel_buffer(320, 200);
+/// low_level_render(pixel_buffer.width(), pixel_buffer.height(),
+///                  static_cast<unsigned char *>(pixel_buffer.begin()));
+/// slint::Image image(pixel_buffer);
+/// ```
+///
+/// Another use-case is to import existing image data into Slint, by
+/// creating a new Image through copying the buffer.
+///
+/// ```cpp
+/// slint::Image image(slint::SharedPixelBuffer<slint::Rgb8Pixel>(the_width, the_height,
+///     static_cast<slint::Rgb8Pixel*>(the_data));
+/// ```
+///
+/// This only work if the static_cast is valid and that the underlying data has the same
+/// memory layout as slint::Rgb8Pixel or slint::Rgba8Pixel. Otherwise, you will have to do a
+/// pixel conversion as you copy the pixels
```

Comment:
```suggestion
/// This only works if the static_cast is valid and the underlying data has the same
/// memory layout as slint::Rgb8Pixel or slint::Rgba8Pixel. Otherwise, you will have to do a
/// pixel conversion as you copy the pixels:
```